### PR TITLE
change jobs names for child jobs in phonopy

### LINKED ISSUE #982: phonopy.py creates jobs with always the same name
https://github.com/pyiron/pyiron/blob/8e3e7678b50ceebd106bdd341e19d4990988f495/pyiron/atomistics/master/phonopy.py#L97

I created two phonopy jobs with different structures. These jobs have different names, but the child jobs are both called `supercell_phonon_0`, which makes it impossible to load by job name. Can I change it to something like `"supercell_{}_{}".format(self._job.job_name, ind)`?

### PR DESCRIPTION
**Pull request by samwaseda**
Closes #982

### CODE REVIEW

File: pyiron/testing/randomatomistic.py [hunk 1]
```diff
@@ -461,6 +461,9 @@ def from_hdf(self, hdf=None, group_name=None):
         super(AtomisticExampleJob, self).from_hdf(hdf=hdf, group_name=group_name)
         self._structure_from_hdf()
 
+    def run_static(self):
```
Comment (jan-janssen) [inline pyiron/testing/randomatomistic.py:464]: I do not think this change belongs to this pull request.

Comment (samwaseda) [reply pyiron/testing/randomatomistic.py:464]: That's true - now I'm going to open another pull request

**Comment by samwaseda** [comment]
I don't understand why murnaghan tests started to fail...

**Comment by jan-janssen** [comment]
> I don't understand why murnaghan tests started to fail...

I would assume it is related to your `run_static()` function.

**Comment by samwaseda** [comment]
> I would assume it is related to your `run_static()` function.

I tested it with and without `run_static` but somehow it fails anyway...

**Comment by samwaseda** [comment]
... and without `run_static`, `test_phonopy.py` fails.

**Comment by jan-janssen** [comment]
The phonopy job needs forces, but the basic test job only has energy but no forces.

**Comment by samwaseda** [comment]
Nice catch!!

**Comment by jan-janssen** [comment]
It should be working now - but the performance is slow as phonopy struggles calculate the phonons from random forces

**Comment by samwaseda** [comment]
> It should be working now - but the performance is slow as phonopy struggles calculate the phonons from random forces

Just remove `run`. I'm gonna replace it by the new Einstein class.

**Comment by yury-lysogorskiy** [comment]
I mean to access subjobs, instead `project.load(sub_job_name)` do something like 
```python
j1=project['unique_master_job_name_1/supercell_phonon_0']
j2=project['unique_master_job_name_2/supercell_phonon_0']
```
in your case, that triggered this PR

**Comment by samwaseda** [comment]
> I mean to access subjobs, instead `project.load(sub_job_name)` do something like
> 
> ```python
> j1=project['unique_master_job_name_1/supercell_phonon_0']
> j2=project['unique_master_job_name_2/supercell_phonon_0']
> ```
> 
> in your case, that triggered this PR

Sure, that's also possible. But that's rather a question of style. I presume you had a practical problem, right? Can we maybe talk about that?

Talking about the concept: we're anyway generally going away from the notion of Master and Child, because we start to see cases where it's not really clear what is the master/child of what & we realised that there's less flexibility with this hierarchy (like when we want to change the project location, because in this case not only you have to change the locations of jobs, but also we'd have to take care of their sub jobs). It's true that some of the oldest codes such as Murnaghan or Phonopy the job folders could be within other job folders, but it might be changed at some point. This being said, it should anyway be backward compatible, so feel free to post issues whenever it doesn't work.

**Pull request merged**